source/adapters/cuda/event.cpp

```diff
@@ -256,7 +261,22 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventRelease(ur_event_handle_t hEvent) {
     ur_result_t Result = UR_RESULT_ERROR_INVALID_EVENT;
     try {
       ScopedContext Active(hEvent->getContext());
-      Result = hEvent->release();
+    if (!hEvent->backendHasOwnership()) {
+      Result = UR_RESULT_SUCCESS;
+    }
+    else {
+      assert(hEvent->getQueue() != nullptr);
+      assert(hEvent->getContext() != nullptr);
+
+      auto queue = event_ptr->getQueue();
```

Comment:
This will fail for interop. See https://oneapi-src.github.io/unified-runtime/core/api.html#ureventcreatewithnativehandle

Reply:
Thanks, fixed now